### PR TITLE
Buckle parent logic

### LINKED ISSUE #2239: Op table unbuckles AND removes anestesia tank
## Reproduction:
1) Buckle someone into an optable.
2) Apply mask.
3) Unbuckle the person by clicking the table.
4) Watch as you unbuckle, remove the mask _and the tank_. 
image
Also the person doing the untaking isn't correctly shown.

### PR DESCRIPTION
**Pull request by Rohesie**
Makes certain buckles follow parent logic where it was broken.

Fixes #2239 .
Also fixes the unreported bug of powerloader unbuckling of others not behaving correctly for the same reason.
Removes the useless `can_buckle = FALSE` definition from pipes, when that value is already such by default in the parents.

Should substitute #2259 .

### CODE REVIEW

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -135,17 +135,32 @@
...
-			buckled_mob.anchored = initial(buckled_mob.anchored)
-			buckled_mob.update_canmove()
 
-			var/M = buckled_mob
-			buckled_mob = null
-			UnregisterSignal(M, COMSIG_LIVING_DO_RESIST)
-			afterbuckle(M)
+/obj/proc/unbuckle(mob/user, silent = TRUE)
+	if(!buckled_mob || buckled_mob.buckled != src)
+		CRASH("[src] called unbuckle() with [user ? user : "no"] user and [buckled_mob ? (buckled_mob.buckled ? "with [buckled_mob.buckled] as buckled_mob.buckled" : "no buckled_mob.buckled") : "no buckled_mob"].")
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]: Gonna wait a bit before merging this to see if this produces any unwanted runtimes.

Comment (Rohesie) [reply code/game/objects/objs.dm]: It did and they got fixed.

Comment (DominikPanic) [reply code/game/objects/objs.dm]: Still gonna give it a few days, then it can get removed.

Comment (DominikPanic) [reply code/game/objects/objs.dm]: Fuck it, I can live with a single CASH in my life

Comment (Rohesie) [reply code/game/objects/objs.dm]: Remind me to open a PR to remove it.
I was waiting until it had been considered tested enough.

**Pull request merged**